signer/src/stacks/contracts.rs

```diff
@@ -371,16 +383,20 @@ impl CompleteDepositV1 {
     ///    bitcoin blockchain.
     /// 4. Check that the sweep transaction uses the indicated deposit
     ///    outpoint as an input.
-    async fn validate_sweep_tx<C>(&self, ctx: &C, req_ctx: &ReqContext) -> Result<(), Error>
+    /// 9. That the first input into the sweep transaction is the signers'
+    ///    UTXO.
+    async fn validate_sweep_tx<C>(&self, ctx: &C, req_ctx: &ReqContext) -> Result<Amount, Error>
     where
         C: Context + Send + Sync,
     {
         let db = ctx.get_storage();
-        // First we check that we have a record of the transaction.
-        let sweep_tx = db
-            .get_bitcoin_tx(&self.sweep_txid, &self.sweep_block_hash)
-            .await?
-            .ok_or_else(|| DepositErrorMsg::SweepTransactionMissing.into_error(req_ctx, self))?;
+        let rpc = ctx.get_bitcoin_client();
+        // First we check that bitcoin-core has a record of the transaction
+        // where we think it should be.
+        let txid = &self.sweep_txid;
+        let Some(sweep_tx) = rpc.get_tx_info(txid, &self.sweep_block_hash).await? else {
+            return Err(DepositErrorMsg::SweepTransactionMissing.into_error(req_ctx, self));
+        };
```

Comment:
qq: here we rely on bitcoin-core, while usually(?) we seem to take our storage as source of truth; mixing the two could cause some not-so-fun-to-track coherence issues, so why are we not trusting our storage here?

Reply:
Yeah, we usually only reach out to the database for this information. Here I used bitcoin-core because I felt that it was much easier than using the database. Specifically, I had thought that I would need to write and test a few more queries before we could get the information that we need. But, I probably miscalculated and maybe we just need one more query.

And I'm not so worried about coherence issues (well, at least not yet). Here the RPC might succeed while the equivalent queries might fail because we haven't processed the bitcoin block yet (but I'm not worried about that either). However, I am worried about bitcoin-core not computing the "UndoData" that is necessary for the RPC to actually work as intended. So when we have time we should fix this up to use the database.